### ISSUE TITLE
CKV_AWS_33 false positive for deny statements

### ISSUE THREAD

**Issue by arothian** [bridgecrewio/checkov#4394]
**Describe the issue**
CKV_AWS_33 provides false-positives for KMS KeyPolicy statements that have a wildcard principle with `Effect: "Deny"`

**Examples**
```yaml
TheKey:
  Type: AWS::KMS::Key
  Properties:
    KeyPolicy:
      Version: "2012-10-17"
      Statement:
        - Effect: "Deny"
          Principal: "*"
          ...
```

**Version (please complete the following information):**
 - Checkov Version [2.2.327]
